### PR TITLE
Remove deprecated sections support

### LINKED ISSUE #614: Remove Deprecated Sections Support
## Description ##
As a developer, I want to remove the DATABASE, SCREEN, PROGRAM-LIBRARY, and REPORT SECTION sections as they are deprecated.

## Acceptance criteria ##
1. Removed DATABASE SECTION
1. Removed SCREEN SECTION
1. Removed PROGRAM-LIBRARY SECTION
1. Removed REPORT SECTION 

## Definition of Done ##
- The story is implemented according to the description.
- Unit testing done (Preferably by TDD approach, coverage should be at least 70%).
- Code review done by at least one other team member.
- Manual testing is done.
- Documentation is added and reviewed (if needed).
- The pull request is approved.
- The story is merged into the development branch.
- Tests are performed on the development branch.

### PR DESCRIPTION
**Pull request by temanbrcom**
Closes  #614

### CODE REVIEW

File: server/src/test/java/com/broadcom/lsp/cobol/usecases/TestSameCopybookStatementsInDifferentPlacesTreatedAsDifferentEntries.java [hunk 1]
```diff
@@ -18,36 +18,26 @@
...
-          + "       01  {$*PARENT}.\n"
           + "       COPY {~REPL}.\n"
-          + "       PROCEDURE DIVISION.\n"
-          + "       {#*MAINLINE}. \n"
-          + "           MOVE 0 TO {$TAG-ID}.\n"
-          + "           MOVE {_COPY {~REPL}.|1|2|3_}\n"
-          + "           GOBACK. ";
-  private static final String REPL = "       {05|1} {$*TAG-ID|2|3} PIC 9.\n";
+          + "       {_COPY {~REPL}.|1_}\n";
+  private static final String REPL = "       {DATA|1} DIVISION.\n";
```
Comment (grianbrcom) [inline server/src/test/java/com/broadcom/lsp/cobol/usecases/TestSameCopybookStatementsInDifferentPlacesTreatedAsDifferentEntries.java:40]: You have copied this REPL twice in your program, but only the second copy contains an error. That means that you use the same `1` diagnostic tag in two places. I expect that the test fails, but it is not.

**Review by grianbrcom — APPROVED** [review]
Looks good, but suspicious :mag:

File: server/src/test/java/com/broadcom/lsp/cobol/usecases/TestSameCopybookStatementsInDifferentPlacesTreatedAsDifferentEntries.java (hunk 1 above)
Comment (grianbrcom) [reply server/src/test/java/com/broadcom/lsp/cobol/usecases/TestSameCopybookStatementsInDifferentPlacesTreatedAsDifferentEntries.java:40]: A-ha! I got it—the error related to copybook, not extended document. Now I see that it's OK.

**Review by asatklichov — APPROVED** [review]
🔢

File: server/src/test/java/com/broadcom/lsp/cobol/usecases/TestSameCopybookStatementsInDifferentPlacesTreatedAsDifferentEntries.java (hunk 1 above)
Comment (temanbrcom) [reply server/src/test/java/com/broadcom/lsp/cobol/usecases/TestSameCopybookStatementsInDifferentPlacesTreatedAsDifferentEntries.java:40]: Yes, that's correct. The reason why I changed this test was just that we are going to support a lot of new semantic features, so most likely the number of errors, as well as their messages, will change frequently. Here I just check that the different copy statements represent different copybook entries, so the test semantic remains the same after the changes.

**Pull request merged**